packages/core/botpress/src/util.js

```diff
@@ -123,6 +126,44 @@ const getPackageName = pkg => {
   }
 }
 
+const existVersion = version => {
+  if (version === undefined) {
+    throw "Version doesn't exist in botfile.js"
+  }
+
+  return version
+}
+const isValidString = version => {
+  if (_.isEmpty(version) || !_.isString(version)) {
+    throw 'Version must be non-empty string'
+  }
+
+  return version
+}
+const isValidFormat = version => {
+  try {
+    // TODO: change this method if "semver" module will implement semver.isValid()
+    semver.valid(version)
+  } catch (err) {
+    throw 'Version must have semver format (e.g. 10.25.0)'
+  }
+
+  return version
+}
+const isEqualOrGreater = majorVersion => version => {
+  const msg = `Your bot may be incompatible with botpress v${majorVersion}
```

Comment:
I'm not sure it implements exactly the checks I've mentioned in the ticket: `Check that the major version of bp is exactly the same as of the version from the botfile. Also check that the version of bp is greater than or equal to the version from the botfile.`

I may be wrong but I feel the part about `Check that the major version of bp is exactly the same as of the version from the botfile` will not hold, like if botfile is `v10.25.0` and botpress is `11.0.0` it will pass the check